lib/Source/AutoHog/AutoHogImporter.cpp

```
@@ -0,0 +1,516 @@
+        assert(table.IsArray() && "Expected 'tableT' to be an array");
+        int lookupTable = parseLut(table.GetArray());
+        lookupTables.push_back(lookupTable);
+      }
+      op = builder.create<cggi::MultiLutLinCombOp>(
+          funcOp.getLoc(), resultTypes, operands, coefficients,
+          builder.getDenseI32ArrayAttr(lookupTables));
+    } else if (strcmp(cellType, "HomGateS") == 0) {
+      SmallVector<int> coefficients;
+      coefficients.resize(operands.size());
+      parseLinCombCoefficients(cell["weights"].GetObject(),
+                               inputPortNameToOperandIndex, coefficients);
+      // Only one output port, so just take the first LUT
+      auto lookupTableJson = cell["tableT"].MemberBegin()->value.GetArray();
+      int lookupTable = parseLut(lookupTableJson);
+      op = builder.create<cggi::LutLinCombOp>(
+          funcOp.getLoc(), resultTypes, operands, coefficients,
+          builder.getI32IntegerAttr(lookupTable));
+    } else if (strcmp(cellType, "AND") == 0) {
+      op = builder.create<cggi::AndOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "NAND") == 0) {
+      op = builder.create<cggi::NandOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "NOR") == 0) {
+      op = builder.create<cggi::NorOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "OR") == 0) {
+      op = builder.create<cggi::OrOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "XOR") == 0) {
+      op = builder.create<cggi::XorOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "XNOR") == 0) {
+      op = builder.create<cggi::XNorOp>(funcOp.getLoc(), resultTypes, operands);
+    } else if (strcmp(cellType, "NOT") == 0) {
+      op = builder.create<cggi::NotOp>(funcOp.getLoc(), resultTypes, operands);
+    } else {
+      assert(false && "Unknown cell type");
```

in release mode we would skip this assertion, should we actually return a failure? or maybe at least a debug message?